objects/src/transaction/proven_tx.rs

```diff
@@ -0,0 +1,111 @@
+use super::{AccountId, ConsumedNoteInfo, CreatedNoteInfo, Digest, Felt, Hasher, Vec};
+use miden_core::{ProgramInfo, StackInputs, StackOutputs};
+use miden_verifier::{verify, ExecutionProof, VerificationError};
+
+/// Resultant object of executing and proving a transaction. It contains the minimal
+/// amount of data needed to verify the transaction was executed correctly.
+/// Contains:
+/// - account_id: the account that the transaction was executed against.
+/// - initial_account_hash: the hash of the account before the transaction was executed.
+/// - final_account_hash: the hash of the account after the transaction was executed.
+/// - consumed_note_info: a list of consumed notes.
+/// - created_note_info: a list of created notes.
+/// - tx_script_root: the script root of the transaction.
+/// - block_ref: the block hash of the last known block at the time the transaction was executed.
+/// - proof: the proof of the transaction.
+pub struct ProvenTransaction {
+    account_id: AccountId,
+    initial_account_hash: Digest,
+    final_account_hash: Digest,
+    consumed_note_info: Vec<ConsumedNoteInfo>,
+    created_note_info: Vec<CreatedNoteInfo>,
+    tx_script_root: Digest,
```

Comment:
`tx_script_root` should be optional (i.e., `Option<Digest>`) as not all transactions need to have it.

Reply:
updated